setup.ps1

```diff
@@ -1,71 +1,139 @@
 param (
-    [string]$oracleContainerName = "psw-oracle-1",
-    [string]$connectionStringName = "OracleConnectionString",
-    [string]$tagName = "setup-oracle-action"
+    [string]$OracleContainerName = "psw-oracle-1",
+    [string]$StorageContainerName = "psworacle1",
+    [string]$ConnectionStringName = "OracleConnectionString",
+    [string]$TagName = "setup-oracle-action",
+    [string]$InitScript = $null
 )
 
 $dockerImage = "gvenzl/oracle-xe:21-slim"
 $oraclePassword = "Welcome1"
-$ip = "127.0.0.1"
+$ipAddress = "127.0.0.1"
 $connectionPort = 1521
 $runnerOs = $Env:RUNNER_OS ?? "Linux"
 $resourceGroup = "GitHubActions-RG"
 $healthCheckCommand = ""
+$scriptExecutionCommand = ""
 
-Write-Output "::add-mask::$ip"
+Write-Output "::add-mask::$ipAddress"
 
 if ($runnerOs -eq "Linux") {
     Write-Output "Running Oracle using Docker"
-    docker run --name "$($oracleContainerName)" -d -p "$($connectionPort):$($connectionPort)" -e ORACLE_PASSWORD=$oraclePassword $dockerImage
+    docker run --name "$($OracleContainerName)" -d -p "$($connectionPort):$($connectionPort)" -e ORACLE_PASSWORD=$oraclePassword $dockerImage
 
-    $healthCheckCommand = "docker exec ""$($oracleContainerName)"" ./healthcheck.sh XEPDB1"
+    $healthCheckCommand = "docker exec ""$($OracleContainerName)"" sqlplus system/$($oraclePassword)@$($ipAddress):$($connectionPort)/XEPDB1"
+    $scriptExecutionCommand = "Get-Content $($InitScript) | docker exec -i ""$($OracleContainerName)"" sqlplus system/$($oraclePassword)@$($ipAddress):$($connectionPort)/XEPDB1"
 }
 elseif ($runnerOs -eq "Windows") {
     $hostInfo = curl -H Metadata:true "169.254.169.254/metadata/instance?api-version=2017-08-01" | ConvertFrom-Json
     $region = $hostInfo.compute.location
     $runnerOsTag = "RunnerOS=$($runnerOs)"
-    $packageTag = "Package=$tagName"
+    $packageTag = "Package=$TagName"
+    $dateTag = "Created=$(Get-Date -Format "yyyy-MM-dd")"
+    $mountPath = "/mnt/scripts";
+
+    Write-Output "Creating a storage account (This can take a while.)"
+    $storageAccountDetails = az storage account create --name $StorageContainerName --location $region --resource-group $resourceGroup --sku Standard_LRS | ConvertFrom-Json
```

Comment:
Do we need the storage account if there is no script to execute?

Reply:
@andreasohlund yes we do. Unfortunately, we have to upload a special health check script to properly discover whether the container is really up and running since ACI container exec would ignore arguments we pass to the sql plus execution.

Reply:
Here is what we could do: because we run Linux containers we could use git repo mounts and add the health check script to the repo here. Then when there is no init script we could mount this repo which will then be cloned into the container. That allows us to execute the health script from the repo without having to create the storage account. That might save us time but only as long as cloning this repo including the history is faster

https://docs.microsoft.com/en-us/azure/container-instances/container-instances-volume-gitrepo

Reply:
We decided not to pursue the github repository route. We feel it is cleaner to always use the same approach.